### ISSUE TITLE
Support to Capybara.server= with a proc was removed without deprecation

### ISSUE THREAD

**Issue by rafaelfranca** [teamcapybara/capybara#2000]
## Meta
Capybara Version:
3.0.0

## Expected Behavior

```ruby
@old_server = Capybara.server
Capybara.server = @old_server
```

To work

## Actual Behavior

```ruby
@old_server = Capybara.server
Capybara.server = @old_server
```

This raise an error because `@old_server` is a proc and `server=` calls `to_sym` in a proc.

This behavior was remove in https://github.com/teamcapybara/capybara/commit/ba7674086cbcd3b22d3614011815bc5d483e5960 because `server` with a block was deprecated, but `server=` with a proc was not. Without his behavior it is not possible to reset the capybara server to the previous state correctly.

**Comment by twalpole** [comment]
@rafaelfranca Yeah - this was an oversight.  The functionality for `server=` will be restored in 3.0.1 later today.

**Comment by rafaelfranca** [comment]
Thank you so much! That will make the rails test suite green again

**Comment by twalpole** [comment]
@rafaelfranca Fixed in 3.0.1 - thanks for the heads up
